docs/DeveloperGuide.adoc

```diff
@@ -36,19 +36,95 @@ This developer guide specifies the technical and non-technical details of THRIFT
 
-_/src/java/thrift/logic/Logic.java_:: Italic text indicates that it is a file path.
+_/src/java/thrift/logic/Logic.java_:: Italic text indicates that it is a file/github path.
 
-`currentMonth`:: Text with grey highlight (called a mark-up) indicates that it is a method/variable name.
+`currentMonth`:: Text with grey highlight (called a mark-up) indicates that it is a method/variable/setting parameter name.
 
 `**THRIFT**`:: Bold text with mark-up indicates a class/package name.
 
+`__Model.java__`:: Italic text with mark-up indicates a file name.
+
 == Setting up
 
-Refer to the guide <<SettingUp#, here>>.
+In this section, we will be introducing the way to set up THRIFT.
+
+=== Prerequisites
+
+. JDK 11 or above
+. IntelliJ IDE
++
+[NOTE]
+IntelliJ by default has Gradle and JavaFx plugins installed. +
+Do not disable them. If you have disabled them, go to *File* > *Settings* > *Plugins* to re-enable them.
+
+=== Setting up the project in your computer
+
+. Fork this repo, and clone the fork to your computer
+. Open IntelliJ (if you are not in the welcome screen, click *File* > *Close Project* to close the existing project dialog first)
+. Set up the correct JDK version for Gradle
+.. Click *Configure* > *Project Defaults* > *Project Structure*
+.. Click *New...* and find the directory of the JDK
+. Click *Import Project*
+. Locate the `__build.gradle__` file and select it. Click *OK*
+. Click *Open as Project*
+. Click *OK* to accept the default settings.
+
+=== Verifying the setup
+
+. Run the `*thrift.Main*` and try a few commands
+. <<Testing,Run the tests>> to ensure they all pass.
+
+=== Configurations to do before writing code
+
+==== Configuring the coding style
+
+This project follows https://github.com/oss-generic/process/blob/master/docs/CodingStandards.adoc[oss-generic coding standards]. IntelliJ's default style is mostly compliant with ours but it uses a different import order from ours. To rectify,
+
+. Go to *File* > *Settings...* (Windows/Linux), or *IntelliJ IDEA* > *Preferences...* (macOS)
+. Select *Editor* > *Code Style* > *Java*
+. Click on the *Imports* tab to set the order
+
+* For `Class count to use import with '\*'` and `Names count to use static import with '*'`: Set to *999* to prevent IntelliJ from contracting the import statements
+* For `Import Layout`: The order is *import static all other imports*, *import java.\**, *import javax.**, *import org.\**, *import com.**, *import all other imports*. Add a *<blank line>* between each *import*
+
+Optionally, you can follow the <<UsingCheckstyle#, UsingCheckstyle.adoc>> document to configure Intellij to check style-compliance as you write code.
+
+==== Updating documentation to match your fork
+
+After forking the repo, the documentation will still have the SE-EDU branding and refer to the _se-edu/addressbook-level3_ repo.
+
+If you plan to develop this fork as a separate product (i.e. instead of contributing to _se-edu/addressbook-level3_), you should do the following:
```

Comment:
I think we might need to change this to THRIFT context. If the developer is to continue building onto our application.